modules/ROOT/pages/migration-from-5x.adoc

```diff
@@ -3,88 +3,614 @@
+
+Instead, the `mceAddEditor` and `mceToggleEditor` commands use `id` and `options` values from an object passed when the command is run.
+
+For example —
+
+`tinyMCE.execCommand('mceAddEdit', false, { id: 'ed-1', options: { readonly: false, toolbar: 'bold italic strikethrough' } })`
+
+— finds a `textarea` with id `ed-1` and turns it into a working {productname} editor that displays the _bold_, _italic_, and _strikethrough_ buttons in the editor toolbar.
+
+
+[[options]]
+== Options
+
+=== `element_format`
+
+Previously, `element_format` had no explicit default value. It’s implicit value, however, was `xhtml`.
+
+In {productname} 6.0, `element_format` now has an explicit default value: `html`.
+
+This new, and explicit default changes how entered material is serialised. In particular, self-closing elements, such as `<hr>`, `<br>`, and `<img>` no longer present in the `xhtml` standard fashion, as `<hr />`, `<br />`, and `</img />` respectively.
+
+NOTE: If `xhtml` standard representation is required, explicitly setting `element_format` to `xhtml` works as before.
+
+=== `image_upload_handler`
+
+Previously the `images_upload_handler` option passed `success` or `failure` callbacks to report the status of an image upload. In {productname} 6, `images_upload_handler` reports upload results by returning a `Promise`, with a value of either `resolve` or `reject`.
+
+=== `link_default_protocol`
+
+The `link_default_protocol` option sets the link protocol used by links added or edited using the _link_ dialog.
+
+It now defaults to `https`. Previously it defaulted to `http`.
+
+NOTE: The `link_default_protocol` value is only applied to an edited or inserted link if the protocol is not explicitly specified.
+
+[[text-patterns]]
+== Text Patterns
+
+As noted in the xref:6.0-release-notes.adoc#removed-or-deprecated-options-textpattern[_{productname} 6.0 Release Notes_], the `textpattern_patterns` option:
+
+* was renamed to `text_patterns`;
+* and had its functionality moved to the Core of {productname} 6.0.
+
+As well, the `textpattern` API has been removed.
+
+Finally, and unlike in previous versions of {productname}, text patterns are now on by default. `text_patterns: false` turns the functionality off.
+
+NOTE: Using `text_patterns: []` to turn `text_patterns` off is not supported. It sets `text_patterns` to an empty array. Having, consequently, no patterns to match, it presents as if `text_patterns` is off. Appearance is not reality, however. Setting `text_patterns: false` is the only supported way of disabling `text_patterns`.
+
+After upgrading:
+
+* rename any options in your {productname} init configuration to match the new name.
+* remove `textpattern` from your plugins list.
+
+=== tinymce.activeEditor.options.get
+
+Previously `tinymce.activeEditor.options.get('plugins')` returned the configured plugins as a string.
+
+As of {productname} 6, `tinymce.activeEditor.options.get('plugins')` now returns the configured plugins as an array.
```

Comment:
As noted in the options page, this isn't quite correct. So going to copy Millie's suggestion from there:
```suggestion
Previously, the list of configured plugins could be retrieved by calling `editor.getParam('plugins')` and returned the configured plugins as a string.

As of {productname} 6, the list of configured plugins is retrieved by calling `editor.options.get('plugins')` and now returns the configured plugins as an array.
```